### PR TITLE
feat(policy): implement get poap's policy

### PR DESCRIPTION
**Pull request by mthinh**
# What ?
- Implement get poap policy
- Show policy requirements when user click in view requirements in poap detail

# Screenshot
Screenshot 2023-08-29 at 07 47 01

### CODE REVIEW

File: lib/core/presentation/widgets/poap/poap_policy_popup/nodes/poap_policy_twitter_node_widget.dart [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  final bool result;
+
+  @override
+  Widget build(BuildContext context) {
+    final colorScheme = Theme.of(context).colorScheme;
+    final t = Translations.of(context);
+    final errorNode = node.children?.firstWhere((nodeItem) => nodeItem.value == 'error');
+    final error = errorNode?.children?.isNotEmpty == true ? errorNode?.children![0].value ?? '' : '';
+    String? errorText;
+    // TODO: var twitterNotConnected = false;
```
Comment (sieuhuflit) [inline lib/core/presentation/widgets/poap/poap_policy_popup/nodes/poap_policy_twitter_node_widget.dart:25]: 👍

File: lib/core/presentation/widgets/poap/poap_policy_popup/poap_policy_node_widget.dart [hunk 2]
```diff
@@ -0,0 +1,190 @@
+// ignore_for_file: use_if_null_to_convert_nulls_to_bools, avoid_bool_literals_in_conditional_expressions
```
Comment (sieuhuflit) [inline lib/core/presentation/widgets/poap/poap_policy_popup/poap_policy_node_widget.dart:1]: Can you fix even ignore

File: lib/core/presentation/widgets/poap/poap_policy_popup/poap_policy_node_widget.dart [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+      for (final nodeItem in node!.children!) {
+        if (nodeItem.value == 'result') {
+          result = nodeItem.children!.isNotEmpty ? nodeItem.children![0].value == 'true' : false;
+        } else {
+          excludedResultNodes.add(nodeItem);
+        }
+      }
+
+      if (node!.value == 'and') {
```
Comment (sieuhuflit) [inline lib/core/presentation/widgets/poap/poap_policy_popup/poap_policy_node_widget.dart:40]: These and or should be constants, seem like use more than 1 time

File: lib/core/presentation/widgets/poap/poap_policy_popup/poap_policy_node_widget.dart [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+                    ),
+                ],
+              );
+            }).toList(),
+          ),
+        );
+      }
+    }
+
+    if (node!.value == 'user-geolocation' && node?.children?.isNotEmpty == true) {
```
Comment (sieuhuflit) [inline lib/core/presentation/widgets/poap/poap_policy_popup/poap_policy_node_widget.dart:105]: Should we define some enum for these node.value

**Comment by mthinh** [comment]
@sieuhuflit pls help recheck

**Review by sieuhuflit — APPROVED** [review]

**Pull request merged**